support/elasticsearch2_search.erl

```diff
@@ -721,6 +751,43 @@ filter_categories(Cats, Context) ->
         Cats
     ).
 
+% Add wildcards to the text unless there are operators.
+add_wildcards(Text) ->
+    Text1 = z_convert:to_binary(Text),
+    case is_with_operators(Text1) of
+        false ->
+            {iolist_to_binary(lists:join(32, add_wildcards_1(Text1))), prefix};
+        true ->
+            {Text1, default}
+    end.
+
+%% @doc Check if the text contains operators. This is a bit of a heuristic, as it
+%% is not always clear if a '-' is just a normal character in a name or an operator.
+%% https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html
+is_with_operators(<<>>) -> false;
+is_with_operators(<<" ", T/binary>>) -> is_with_operators(T);
+is_with_operators(<<"\"", _/binary>>) -> true;
+is_with_operators(<<"-", _/binary>>) -> true;
+is_with_operators(<<"+", _/binary>>) -> true;
+is_with_operators(S) ->
+    binary:match(S, <<"\"">>) =/= nomatch
+    orelse binary:last(S) =:= $*
+    orelse re:run(S, <<" [-+][^ ]">>) =/= nomatch
+    orelse re:run(S, <<" \\+ ">>) =/= nomatch
+    orelse re:run(S, <<" (AND|OR) ">>) =/= nomatch
+    orelse re:run(S, <<"[^ ]\\~[0-9]">>) =/= nomatch
+    orelse re:run(S, <<"[^ ]\\* ">>) =/= nomatch.
```

Comment:
A more readable name would be `has_operators` - I'm not in favor of always just adding is_ for booleans; would prefer taking predicate names, and nouns (with or without "get_") for non-boolean values. (We can update the code in a later iteration.)